server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/NodeAllocationOrdering.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.cluster.routing.allocation.allocator;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class NodeAllocationOrdering {
```

Comment:
Can we add javadoc on the purpose of this class? Seems to be something around round-robin order of allocation but would be good to document it here.

Reply:
:+1: done in c5994780f15b578b4176a43bf64b4e85591edc08.